x/shared/types/service.go

```diff
@@ -0,0 +1,10 @@
+package types
+
+// NewService creates a new Service instance
```

Comment:
1. Do you think this is really necessary? Seems like an unnecessary abstraction (for now) given how it maps 1:1.
2. If we do decide to go down this path, add a `TODO_TECHDEBT` so we adopt it everywhere else for consistency (not in this PR)

Reply:
@Olshansk the problem right now is that every time we create a new service instance, we create a literal. Going forward, that makes it very difficult to make updates and identify issues. By using a constructor, it allows us to easily track usages and if we change how we construct them it immediately and clearly breaks the parts of the code that require updating. Constructing literals masks issues because Go automatically adds default values to any property not specified when constructing the literal (and that is part of why Go best practices say to use constructor functions rather than literals). I added the `service.go` file here because everywhere else that we have constructor functions for protobuf messages we have a go file right next to the generated proto file (e.g. `params.pb.go` has a params constructor function in `params.go`). Rather than remove the Service constructor, I would suggest we update every place where we construct a new Service instance to use it. I would be happy to do that and I think it would make the code much more robust. As of now, any of those constructed services in tests have a default `0` value for compute units per relay and that could cause unexpected behavior in the future. If you don't want the constructor here, I could move it into service rather than shared, but that would break with the convention I mentioned above.

Reply:
Makes sense to me. Please create a GitHub Issues capturing this and add a TODO_TECHDEBT to make sure we adopt it elsewhere.

`TODO_TECHDEBT(#XXX): ...`

Reply:
@rBurgett Quick feedback on large block of texts like ☝️ 

I appreciate the thoughtful feedback, but just looking at it is a lof of **cognitive load for a reviewer**.

Requests:
- Try to keep it shorter (if possible)
- Use bullet points (if possible)
- Put yourself in the shoes of the person doing the review

🙏